### PR TITLE
Source map support

### PR DESCRIPTION
**Pull request by ry**
Depends on #421

### CODE REVIEW

**Comment by piscisaureus** [comment]
I'm running into some issues: https://gist.github.com/piscisaureus/0103c081d3e3ccf50c1f4587212f3ecd#file-gistfile1-txt

**Comment by ry** [comment]
Seems like you don’t have @types/node ?

**Comment by ry** [comment]
@piscisaureus Maybe your symlinks are messed up and it doesn't create //node_modules or //out/debug/node_modules ? Otherwise I'm not sure...

File: js/v8_source_maps.ts [hunk 1]
```diff
@@ -2,8 +2,9 @@
 // Copyright 2018 the Deno authors. All rights reserved. MIT license.
 // Originated from source-map-support but has been heavily modified for deno.
 import { SourceMapConsumer, MappedPosition } from "source-map";
+import { RawSourceMap } from "source-map";
 import * as base64 from "base64-js";
```
[the remark this answers is not in the document]
Comment (piscisaureus) [reply js/v8_source_maps.ts]: I don't think btoa is available - it's DOM api (node doesn't have it either).

**Comment by piscisaureus** [comment]
> @piscisaureus Maybe your symlinks are messed up and it doesn't create //node_modules or //out/debug/node_modules ? Otherwise I'm not sure...

Symlinks are fine. Running tsc directly also no problem, looks like something related to either ninja or rollup. Will look into it.

**Comment by ry** [comment]
I have removed the asm optimization from this PR. I will submit separately.

**Comment by ry** [comment]
closed in favor of #429

**Pull request closed without merging**